Cleanup quotes a bit

wom opened the pull request:
Just a quick and dirty cleanup of the quotes file.  I did chunk it by commit to make it readable/obvious what is going on.

wom committed:
refactor(quotes): normalize formatting.

wom committed:
feat(quotes): Sort quotes by author.

wom committed:
fix(quotes): remove duplicate quotes and fix errors

- Remove around 30 duplicate quotes with nearly identical wording
- Fix incorrectly formatted Yoda quote with "l='r'" property
- Add a quote from Denis Diderot

RileyGabrielson commented:
Hey thank you for this! Really appreciate the PR

I actually scrambled the order of the quotes so that the daily quote feature wouldn't show a bunch of quotes by the same person in a row. If we want an alphabetized list, we will need another mechanism for ordering. An array of indexes maybe?

wom commented:
Good call, I hadn't looked at that implementation as to why the order was ~random.  

What about this?  I tried grabbing a random index in the table seeded off the current day.  Just testing by incrementing get_days() it appears to do what we want and lets us organize the hardcoded quote list.  I can add to this PR if you think it's worthwhile.

```diff
diff --git a/lua/inspire/main.lua b/lua/inspire/main.lua
index 227763c..526c646 100644
--- a/lua/inspire/main.lua
+++ b/lua/inspire/main.lua
@@ -80,8 +80,27 @@ function main.center_text(text, author, window_width, window_height, word_width)
 	return vertically_centered_lines
 end
 
+local function get_deterministic_daily_index(total_count)
+	-- Use current day as seed for the random generator.
+	local day_number = get_days()
+	-- Create a new seed for each cycle through all quotes
+	local cycle_number = math.floor(day_number / total_count)
+	local seed = day_number + (cycle_number * 10000)
+
+	-- Create deterministic pseudo-random sequence for this day
+	math.randomseed(seed)
+
+	-- Get a deterministic "random" index between 1 and total_count
+	local index = math.random(1, total_count)
+
+	-- Restore random seed
+	math.randomseed(os.time())
+
+	return index
+end
+
 function main.get_quote(config)
-	local index = (get_days() % #config.quotes) + 1
+	local index = get_deterministic_daily_index(#config.quotes)
 	if config.mode == "random" then
 		index = math.ceil(math.random() * #config.quotes)
 	end
```

RileyGabrielson commented:
Love it, yeah this is a great fix

wom committed:
feat(quotes): implement deterministic daily quotes

Add a new function to select quotes deterministically based on the
current day. This ensures a different quote is shown each day, but the
same quote will always appear on the same date. The implementation uses
the day number to create a seed for selecting quotes, ensuring all
quotes are shown before repeating the cycle.

wom commented:
Added new daily-selection logic.
